### DISCUSSION TITLE
Group image messages

### DISCUSSION THREAD

**Discussion by felixgabler** [flyerhq/flutter_chat_ui#234] [Ideas & Requests]
What do you think of adding an option to group all image messages that have not been interrupted by other messages within one message? This would be super helpful for when the user bulk uploads them as to not spam the whole chat window.

I really like signals design for it but perhaps we should also add a builder option for this.

**Comment by demchenkoalex** [comment]
I remember this was a huge deal to implement when we were designing the chat, so we opted out to single messages even without the possible caption. Having no like "specific" backend it was really hard to design upload of multiple files at the same time, that will work with anything. Maybe we should browse some open source packages and see if there is something we might find useful for this.

**Reply by felixgabler** [reply]
I think we can still keep it backend agnostic. All we need is to allow the `ImageMessage` to have multiple images and then render them in a grid and extend the gallery to handle more, right? Actually creating these messages would still be agnostic as it is currently done. People can do whatever they want on `onAttachmentPressed`.

**Reply by demchenkoalex** [reply]
Was thinking more about loading states and stuff, for example now when uploading image we did an easy way - we just show loading indicator on the attachment button. Ideally, loading (uploading, maybe even with progress) will be shown on top of image, i.e. we show image from the local source and progress on top and then replacing local image with the remote url. But I wasn't sure how to optimally implement that. If done correctly, it would allow to show progresses on every message in the grid, showing progress.
